### ISSUE TITLE
Allow to suppress properties same as functions

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#4930]
## Context
I'm trying to configure Detekt using suppressors to allow `lateinit` for only specific names.
For example in all my tests, I use `private lateinit var sut: SystemUnderTest`, so I would like to allow `lateinit` usage on this specific name: `sut`.

## Current Behavior
I tried the following
```yml
potential-bugs:
  LateinitUsage:
    excludes: [ ] # Enforce this for production and test code.
    ignoreFunction:
      - 'sut'
      - 'getSut'
```
but it didn't work.

## Expected Behavior
A way to do this.

---

Note: next step I would like to also allow patterns `^mock.*` and `^fixt.*` which are also properties in test code. Although they're also annotated so I have a way.

**Comment by cortinico** [comment]
> I'm trying to configure Detekt using suppressors to allow `lateinit` for only specific names.

I feel like that this has to be a specific configuration of the rule rather than a generic suppressor 🤔

**Comment by TWiStErRob** [comment]
This is just a specific example, in general treating `fun`s and `val`s the same level is necessary as they're capable of the same things (wrt to scoping).

There are a lot of rules (in performance, potential-bugs, style) that would benefit from being able to be suppressed by project specific naming.

**Comment by BraisGabin** [comment]
A property is like a function in a low-level. But we don't work at that low-level. `ignoreFunction`

I think that it's time for detekt to give the option to implement your own suppressors. I'm going to create an issue to track that.

> There are a lot of rules (in performance, potential-bugs, style) that would benefit from being able to be suppressed by project specific naming.

Could you elaborate a bit more? I'm thinking about a suppressor that takes `Finding.entity.ktElement?.text` and looks for a match in a list of `String`s that you provide from the configuration. But it seems really broad and easy to have a lot of unexpected issues.

**Comment by TWiStErRob** [comment]
> A property is like a function in a low-level. But we don't work at that low-level. ignoreFunction

Yes, I know, the reason I opened this PR is because we don't need to work on that low level, we need a way to treat properties on the same high level as functions. Their PSI is different and therefore function suppressor misses them. To draw a parallel: this lack of support feels like supporting class names in a rule, but not interfaces.

>  I'm thinking about a suppressor that takes Finding.entity.ktElement?.text and looks for a match in a list of Strings that you provide from the configuration.

Wait, am I missing something here? Is the current function suppressor only capable of suppressing fully qualified names? That's a quite hard limitation if that's the case, using `@Suppress` is way easier then. Do you have a real life example of `ignoreFunction`?

**Comment by TWiStErRob** [comment]
Does this need more info, or have reached a conclusion?

**Comment by BraisGabin** [comment]
I don't think so. Anyone can jump in and implement it.
